### PR TITLE
Add dependencies for data loader project

### PR DESCRIPTION
**Pull request by ypeckstadt**
## Description

The following PR adds the dependencies to the shared `build.gradle` file for all data loader sub-projects.

Edit: the ext block has been moved to the main build.gradle ext dependency block.

## Related issues and/or PRs

NA

## Changes made

- Added Lombok dependencies

## Checklist

> The following is a best-effort checklist. If any items in this checklist are not applicable to this PR or are dependent on other, unmerged PRs, please still mark the checkboxes after you have read and understood each item.

- ~~[ ] I have commented my code, particularly in hard-to-understand areas.~~
- ~~[ ] I have updated the documentation to reflect the changes.~~
- ~~[ ] Any remaining open issues linked to this PR are documented and up-to-date (Jira, GitHub, etc.).~~
- ~~[ ] Tests (unit, integration, etc.) have been added for the changes.~~
- [x] My changes generate no new warnings.
- [x] Any dependent changes in other PRs have been merged and published.

## Additional notes (optional)

NA

## Release notes

NA

### CODE REVIEW

File: data-loader/build.gradle [hunk 1]
```diff
@@ -1,9 +1,11 @@
 subprojects {
     group = "scalardb.dataloader"
 
+
     ext {
         apacheCommonsLangVersion = '3.14.0'
         apacheCommonsIoVersion = '2.16.1'
+        lombokVersion = '1.18.34'
```
Comment (komamitsu) [inline data-loader/build.gradle:8]: I noticed `ext` block that defines dependency versions is written in the root `build.gradle`. How about moving this block into it? It has `commonsLangVersion = '3.14.0'` always.

Comment (ypeckstadt) [reply data-loader/build.gradle:8]: Good point. I will use the `commonsLangVersions`. That said, I added the ext in this build.gradle file to separate some dependencies that only apply to the data-loader subprojects. I don't think Lombok will be used in other (sub)projects.

File: data-loader/build.gradle [hunk 2]
```diff
@@ -1,9 +1,11 @@
 subprojects {
     group = "scalardb.dataloader"
 
+
```
Comment (brfrn169) [inline data-loader/build.gradle:4]: Unnecessary empty line?

File: data-loader/build.gradle [hunk 3]
```diff
@@ -23,5 +25,12 @@ subprojects {
         testImplementation "org.mockito:mockito-core:${mockitoVersion}"
         testImplementation "org.mockito:mockito-inline:${mockitoVersion}"
         testImplementation "org.mockito:mockito-junit-jupiter:${mockitoVersion}"
+
```
Comment (brfrn169) [inline data-loader/build.gradle:28]: Ditto. Unnecessary empty line?

**Review by brfrn169 — APPROVED** [review]
Left several minor comments. Other than that, LGTM! Thank you!

**Review by Torch3333 — APPROVED** [review]
LGTM, thank you!

**Pull request merged**